### PR TITLE
[SPARK-50295][DOCS][FOLLOWUP] Document `build-docs` in `docs/README.md`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to document `dev/spark-test-image-util/docs/build-docs` in `docs/README.md`.

### Why are the changes needed?
Add documentation for using script `dev/spark-test-image-util/docs/build-docs`.

### Does this PR introduce _any_ user-facing change?
Yes, Spark developers will be able to know the existence of script `dev/spark-test-image-util/docs/build-docs` through documentation `docs/README.md`.

### How was this patch tested?
Manually check.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @HyukjinKwon @zhengruifeng @LuciferYang @pan3793

File: docs/README.md [hunk 1]
```diff
@@ -126,3 +126,15 @@ To control what API docs get built, you can set any combination of the following
...
+
+## Build docs with docker image (Optional)
+As a Spark developer, you can generate all documents locally as follows:
+
+```sh
+sh dev/spark-test-image-util/docs/build-docs
+```
+
+It will generate all documents on the `container` and `host`.
+Especially when there are conflicts between the libraries required by python development environment
```
Comment (HyukjinKwon) [inline docs/README.md:138]:
```suggestion
Especially when there are conflicts between the libraries required by Python development environment
```

File: docs/README.md [hunk 2]
```diff
@@ -126,3 +126,15 @@ To control what API docs get built, you can set any combination of the following
...
+and the libraries required by generating python docs environment, this is a good choice.
```
Comment (HyukjinKwon) [inline docs/README.md:139]:
```suggestion
and the libraries required by generating Python docs environment, this is a good choice.
```

File: docs/README.md [hunk 3]
```diff
@@ -126,3 +126,15 @@ To control what API docs get built, you can set any combination of the following
...
+Please refer to [SPARK-50295](https://issues.apache.org/jira/browse/SPARK-50295) for details.
```
Comment (HyukjinKwon) [inline docs/README.md:140]: I think no need to link the JIRA here.

File: docs/README.md [hunk 4]
```diff
@@ -126,3 +126,15 @@ To control what API docs get built, you can set any combination of the following
 * `SKIP_RDOC=1`: Skip the R API docs.
 * `SKIP_SQLDOC=1`: Skip the SQL API docs.
 
+
+## Build docs with docker image (Optional)
+As a Spark developer, you can generate all documents locally as follows:
```
Comment (HyukjinKwon) [inline docs/README.md:131]: Let's mention Docker is required.

File: docs/README.md (hunk 3 above)
Comment (panbingkun) [reply docs/README.md:140]: Okay

File: docs/README.md [hunk 5]
```diff
@@ -126,3 +126,15 @@ To control what API docs get built, you can set any combination of the following
 * `SKIP_RDOC=1`: Skip the R API docs.
 * `SKIP_SQLDOC=1`: Skip the SQL API docs.
 
+
+## Build docs with docker image (Optional)
```
Comment (HyukjinKwon) [inline docs/README.md:130]:
```suggestion
## Build docs with docker image (Optional)

```

File: docs/README.md (hunk 4 above)
Comment (panbingkun) [reply docs/README.md:131]: Okay

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise, thanks for fixing this.

**Review by zhengruifeng — APPROVED** [review]

File: docs/README.md [hunk 6]
```diff
@@ -126,3 +126,16 @@ To control what API docs get built, you can set any combination of the following
...
 * `SKIP_SQLDOC=1`: Skip the SQL API docs.
 
+## Build docs with docker image (Optional)
+
+As a Spark developer, you can generate all documents locally as follows:
+
+Note: Before running it, you need to have `docker` installed.
+
+```sh
+$ sh dev/spark-test-image-util/docs/build-docs
```
Comment (pan3793) [inline docs/README.md:136]: sh is not required after granting x permission
```suggestion
$ dev/spark-test-image-util/docs/build-docs
```

Comment (panbingkun) [reply docs/README.md:136]: But it's a bit strange, I granting it as `chmod a+x dev/spark-test-image-util/docs/build-docs`, but it didn't show up in the git logs. Do you know why? thanks.

Comment (panbingkun) [reply docs/README.md:136]: But it's a bit strange, I granting it as `chmod a+x dev/spark-test-image-util/docs/build-docs`, but it didn't show up in the git logs. Do you know why? thanks.
```
(base) ➜  spark-trunk git:(master) ✗ ls -al dev/spark-test-image-util/docs/build-docs
-rw-r--r--@ 1 panbingkun  staff  2794 Nov 29 16:17 dev/spark-test-image-util/docs/build-docs
```

Comment (panbingkun) [reply docs/README.md]: It seems to work, let's take another look this time.

Comment (panbingkun) [reply docs/README.md]: git add --chmod=+x dev/spark-test-image-util/docs/build-docs

**Review by LuciferYang — APPROVED** [review]

**Comment by panbingkun** [comment]
Merged to master.
Thank you, @HyukjinKwon @zhengruifeng @pan3793 @LuciferYang .

**Comment by panbingkun** [comment]
After the pr, the permissions for file `dev/spark-test-image-util/docs/build-docs` are correct.

```shell
(base) ➜  spark-trunk git:(master) ✗ ls -al dev/spark-test-image-util/docs/build-docs
-rwxr-xr-x@ 1 panbingkun  staff  2794 Nov 29 18:59 dev/spark-test-image-util/docs/build-docs
```

**Pull request closed without merging**